src/librbd/internal.h

```diff
@@ -127,6 +127,7 @@ namespace librbd {
   void snap_create_helper(ImageCtx *ictx, Context* ctx, const char *snap_name);
   int snap_list(ImageCtx *ictx, std::vector<snap_info_t>& snaps);
   bool snap_exists(ImageCtx *ictx, const char *snap_name);
+  int snap_exists2(ImageCtx *ictx, const char *snap_name, bool *exists);
```

Comment:
In the internal implementation, just update the ```snap_exists``` signature to return an ```int``` and have ```librbd.cc``` handle the two different APIs by calling the same method.